include/eve/detail/function/simd/x86/friends.hpp

```diff
@@ -9,500 +9,485 @@
-  template<typename T, typename U, typename N>
-  EVE_FORCEINLINE auto self_logor ( sse2_ const&
-                                  , logical<wide<T,N>> v, logical<wide<U,N>> w
-                                  ) noexcept
-  requires( x86_abi<abi_t<T, N>> && x86_abi<abi_t<U, N>> )
+//================================================================================================
+template<typename T, typename U, typename N>
+EVE_FORCEINLINE auto
+self_logor(sse2_ const&, logical<wide<T, N>> v, logical<wide<U, N>> w) noexcept
+    requires(x86_abi<abi_t<T, N>>&& x86_abi<abi_t<U, N>>)
+{
+  if constexpr( !use_is_wide_logical<abi_t<T, N>>::value )
   {
-    if constexpr( !use_is_wide_logical<abi_t<T, N>>::value )
-    {
-      using storage_t = typename logical<wide<T, N>>::storage_type;
-      using m_t       = typename storage_t::type;
-      m_t that = v.storage().value | w.storage().value;
-      return logical<wide<T, N>>(storage_t{that});
-    }
-    else
-    {
-      return self_logor(cpu_{},v,w);
-    }
+    using storage_t = typename logical<wide<T, N>>::storage_type;
+    using m_t       = typename storage_t::type;
+    m_t that        = v.storage().value | w.storage().value;
+    return logical<wide<T, N>>(storage_t {that});
   }
+  else { return self_logor(cpu_ {}, v, w); }
+}
 
-  //================================================================================================
-  template<real_value T, typename N>
-  EVE_FORCEINLINE auto self_lognot(logical<wide<T, N>> v) noexcept
-      requires x86_abi<abi_t<T, N>>
+//================================================================================================
+template<real_value T, typename N>
+EVE_FORCEINLINE auto
+self_lognot(logical<wide<T, N>> v) noexcept requires x86_abi<abi_t<T, N>>
+{
+  if constexpr( !abi_t<T, N>::is_wide_logical )
   {
-    if constexpr( !abi_t<T, N>::is_wide_logical )
-    {
-      using l_t = logical<wide<T, N>>;
-      return l_t{~v.storage()};
-    }
-    else
-    {
-      return bit_cast(~v.bits(), as(v));
-    }
+    using l_t = logical<wide<T, N>>;
+    return l_t {~v.storage()};
   }
+  else { return bit_cast(~v.bits(), as(v)); }
+}
+
+//================================================================================================
+template<real_value T, typename N>
+EVE_FORCEINLINE as_logical_t<wide<T, N>>
+                self_eq(wide<T, N> v, wide<T, N> w) noexcept requires x86_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, N>>();
+  constexpr auto f = to_integer(cmp_flt::eq_oq);
 
-  //================================================================================================
-  template<real_value T, typename N>
-  EVE_FORCEINLINE as_logical_t<wide<T, N>> self_eq(wide<T,N> v, wide<T,N> w) noexcept
-      requires x86_abi<abi_t<T, N>>
+  if constexpr( current_api >= avx512 )
+  {
+    if constexpr( c == category::float32x16 ) return mask16 {_mm512_cmp_ps_mask(v, w, f)};
```

Comment:
I think there was a setting to line up conditions, wasn't there?

Reply:
Can't find it. The one we waited for what that if constexpr was treated as If with respect to else.

Reply:
We will remove the else whenever we run over some to guarantee alignment